Add Kuryr docs about limitation of service without selector (#28022)
openshift/openshift-docs: modules/installation-osp-kuryr-known-limitations.adoc

[id="openstack-general-limitations_{context}"]
== {rh-openstack} general limitations

Using {product-title} with Kuryr SDN has several limitations that apply to all versions and environments:

* `Service` objects with the `NodePort` type are not supported.

> * Clusters that use the OVN Octavia provider driver support `Service` objects for which the `.spec.selector` property is unspecified only if the `.subsets.addresses` property of the `Endpoints` object includes the nodes' or pods' subnet. 

Reviewer: To avoid possessive apostrophes here, I would suggest writing, "...includes the subnet of the nodes or pods" or "includes the node or pod subnet". (ISG mentions this on p. 41.)

> * Clusters that use the OVN Octavia provider driver support `Service` objects for which the `.spec.selector` property is unspecified only if the `.subsets.addresses` property of the `Endpoints` object includes the nodes' or pods' subnet. 

Reviewer 2: ```suggestion
* Clusters that use the OVN Octavia provider driver support `Service` objects for which the `.spec.selector` property is unspecified only if the `.subsets.addresses` property of the `Endpoints` object includes the subnet of the nodes or pods. 
```

> * If the machines subnet is not connected to a router, or if the subnet is connected, but the router has no external gateway set, Kuryr cannot create floating IPs for `Service` objects with type `LoadBalancer`.

Reviewer: Similar to my previous comment, would you want to instead say "the subnet of the machines" or "the machine subnet"? I don't want to pretend that I have any clue. 😜

Reviewer 2: I think it's essentially "the subnet that machines are created on," but it's been a while. @MaysaMacedo, is this the case?

Author: @maxwelldb Yes, it's the subnet that the machines are created on.

> * If the machines subnet is not connected to a router, or if the subnet is connected, but the router has no external gateway set, Kuryr cannot create floating IPs for `Service` objects with type `LoadBalancer`.

Reviewer 2: ```suggestion
* If the subnet on which machines are created is not connected to a router, or if the subnet is connected, but the router has no external gateway set, Kuryr cannot create floating IPs for `Service` objects with type `LoadBalancer`.
```

[id="openstack-general-limitations_{context}"]
== {rh-openstack} general limitations

> Using {product-title} with Kuryr SDN has several limitations that apply to all versions and environments:

Author: The limitations do not apply to all versions and environments.
Topic 2: Services without selectors is only from 4.7+
Topic 3: from 4.6+

Reviewer 2: @MaysaMacedo This will be applied only to 4.7+--the other "version" limitations are to do with RHOSP versions.

Reviewer 2: @MaysaMacedo ^^^

Reviewer 3: Makes sense.

Author: +1

* `Service` objects with the `NodePort` type are not supported.

> * Clusters that use the OVN Octavia provider driver support `Service` objects for which the `.spec.selector` property is unspecified only if the `.subsets.addresses` property of the `EndPoints` object includes the nodes' or pods' subnet. 

Author: Perhaps it's better to leave `Endpoints` here as is the way defined on Kubernetes.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/#endpoints-v1-core

Reviewer 2: @MaysaMacedo As in `Endpoints`? 

```suggestion
* Clusters that use the OVN Octavia provider driver support `Service` objects for which the `.spec.selector` property is unspecified only if the `.subsets.addresses` property of the `Endpoints` object includes the nodes' or pods' subnet. 
```

Reviewer 2: @MaysaMacedo ^^^

Reviewer 3: Yep, it should not be camelcased.

{product-title} with Kuryr SDN does not support `Service` objects with type `NodePort`.

{product-title} with Kuryr SDN and OVN Octavia provider supports `Service` objects

> without specifying `selector` only when the `Endpoints` object `addresses` are either

Reviewer 3: .spec.selector might be better here.
